### DISCUSSION TITLE
Icon request: icon-check-all icon-uncheck-all

### DISCUSSION THREAD

**Discussion by shyndman** [FortAwesome/Font-Awesome#21191] [Icon Requests]
Examples:

![check-all](http://publib.boulder.ibm.com/infocenter/wsdoc400/v6r0/topic/com.ibm.websphere.pmc.iseries.doc/sibuswssecurity/images/wtable_select_all.gif)

![uncheck-all](http://www.fotoview.nl/Graphing/images%5Cuncheckall.png)

I suppose you could get away with stacking the existing icons. I might give that a shot to see how it looks.

**Comment by gramx** [comment]
I think there is a bad link there for one of thoughs here is another example: ![selectall](https://cloud.githubusercontent.com/assets/6108820/10618781/1db3a77c-772f-11e5-9dc5-7ca0acba2091.png)

**Comment by wstrik** [comment]
+1

**Comment by suhuichen** [comment]
+1

**Comment by chellberg** [comment]
+1

**Comment by mafk** [comment]
+1

**Comment by jinglesthula** [comment]
+1 (two squares is better than three, IMO, especially at smaller sizes)

**Comment by dincozdemir** [comment]
+1

**Comment by glueball** [comment]
+1

**Comment by jakubste** [comment]
+1

**Comment by jonatanskogsfors** [comment]
+1

**Comment by clabnet** [comment]
+1

**Comment by ulybu** [comment]
Personally I use this snippet to produce "select all checkbox" :
⚠️ May use pro features

```html
<div style="height: 1em; width: 1em;">
  <i class="far fa-square"            style="position: absolute; top: 0px; left: 0px;"/>
  <i class="fas fa-square fa-inverse" style="position: absolute; top: 2px; left: 2px;"/>
  <i class="far fa-check-square"      style="position: absolute; top: 3px; left: 3px;"/>
  <!--This line for "unselect all" icon
    <i class="far fa-square"      style="position: absolute; top: 3px; left: 3px;"/>
  -->
</div>
```
![image](https://user-images.githubusercontent.com/2479249/37110845-62545162-223e-11e8-8036-07a665df4d84.png)

**Comment by BartusZak** [comment]
Thanks man for provided solution.

JSX Bootstrap v4.0

JSX
```
<div className="selectAllIcons" onClick={() => this.addAllGroups()} title="Wybierz wszystko">
   <i class="fa fa-square"            style={{position: "absolute", top: "0px", left: "0px"}}/>
    <i class="fa fa-square fa-inverse" style={{position: "absolute", top: "2px", left: "2px"}}/>
    <i class="fa fa-check-square"      style={{position: "absolute", top: "3px", left: "3px"}}/>
</div>
 <div className="unSelectAllIcons" onClick={() => this.generatingGroups()} title="Odznacz wszystko">
      <i class="fa fa-square"            style={{position: "absolute", top: "0px", left: "0px"}}/>
      <i class="fa fa-square fa-inverse" style={{position: "absolute", top: "2px", left: "2px"}}/>
       <i class="fa fa-check-square"      style={{position: "absolute", top: "3px", left: "3px"}}/>
      <i class="fa fa-square"            style={{position: "absolute", top: "3px", left: "3px"}}/>
</div>
```

```
CSS
.unSelectAllIcons,
    .selectAllIcons {
        height: 1em;
        width: 1em;
        position: relative;
        display: inline-block;
        margin: 0 20px 0 20px;
}
```
PREVIEW
![image](https://user-images.githubusercontent.com/16143176/37730396-e6a3434a-2d3f-11e8-979e-390d433fea46.png)

**Comment by sensibleworld** [comment]
This is a quick update to let folks know that we’re refining our icon request workflow behind the scenes, and as part of that we’re going through all open requests and filing them into several new buckets to be addressed.

In this case we’ve added this issue to our new list to be created! We can’t promise any firm timelines right now, but this will help us keep track of it.

We’ll update this issue when the icon is released into the wild 🙂
